ID6_data_numtrainex_512
CREATING TFRECORDS
/om/user/sanjanas/eccentricity-data/mnist/
NUM IMAGES VAL: 59488.0
Extracting ../eccentricity-data/mnist/train-images-idx3-ubyte.gz
Extracting ../eccentricity-data/mnist/train-labels-idx1-ubyte.gz
Extracting ../eccentricity-data/mnist/t10k-images-idx3-ubyte.gz
Extracting ../eccentricity-data/mnist/t10k-labels-idx1-ubyte.gz
Data: 0/512
Data: 0/59488
Data: 1000/59488
Data: 2000/59488
Data: 3000/59488
Data: 4000/59488
Data: 5000/59488
Data: 6000/59488
Data: 7000/59488
Data: 8000/59488
Data: 9000/59488
Data: 10000/59488
Data: 11000/59488
Data: 12000/59488
Data: 13000/59488
Data: 14000/59488
Data: 15000/59488
Data: 16000/59488
Data: 17000/59488
Data: 18000/59488
Data: 19000/59488
Data: 20000/59488
Data: 21000/59488
Data: 22000/59488
Data: 23000/59488
Data: 24000/59488
Data: 25000/59488
Data: 26000/59488
Data: 27000/59488
Data: 28000/59488
Data: 29000/59488
Data: 30000/59488
Data: 31000/59488
Data: 32000/59488
Data: 33000/59488
Data: 34000/59488
Data: 35000/59488
Data: 36000/59488
Data: 37000/59488
Data: 38000/59488
Data: 39000/59488
Data: 40000/59488
Data: 41000/59488
Data: 42000/59488
Data: 43000/59488
Data: 44000/59488
Data: 45000/59488
Data: 46000/59488
Data: 47000/59488
Data: 48000/59488
Data: 49000/59488
Data: 50000/59488
Data: 51000/59488
Data: 52000/59488
Data: 53000/59488
Data: 54000/59488
Data: 55000/59488
Data: 56000/59488
Data: 57000/59488
Data: 58000/59488
Data: 59000/59488
Extracting ../eccentricity-data/mnist/train-images-idx3-ubyte.gz
Extracting ../eccentricity-data/mnist/train-labels-idx1-ubyte.gz
Extracting ../eccentricity-data/mnist/t10k-images-idx3-ubyte.gz
Extracting ../eccentricity-data/mnist/t10k-labels-idx1-ubyte.gz
Data: 0/10000
Data: 1000/10000
Data: 2000/10000
Data: 3000/10000
Data: 4000/10000
Data: 5000/10000
Data: 6000/10000
Data: 7000/10000
Data: 8000/10000
Data: 9000/10000
2019-02-08 20:29:59.673264: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2019-02-08 20:29:59.673372: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2019-02-08 20:29:59.673392: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2019-02-08 20:29:59.673408: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2019-02-08 20:29:59.673422: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
2019-02-08 20:29:59.904250: I tensorflow/core/common_runtime/gpu/gpu_device.cc:955] Found device 0 with properties: 
name: GeForce GTX TITAN X
major: 5 minor: 2 memoryClockRate (GHz) 1.076
pciBusID 0000:06:00.0
Total memory: 11.93GiB
Free memory: 367.94MiB
2019-02-08 20:30:00.107170: W tensorflow/stream_executor/cuda/cuda_driver.cc:523] A non-primary context 0x56ee2f0 exists before initializing the StreamExecutor. We haven't verified StreamExecutor works with that.
2019-02-08 20:30:00.107776: I tensorflow/core/common_runtime/gpu/gpu_device.cc:955] Found device 1 with properties: 
name: GeForce GTX TITAN X
major: 5 minor: 2 memoryClockRate (GHz) 1.076
pciBusID 0000:81:00.0
Total memory: 11.93GiB
Free memory: 367.06MiB
2019-02-08 20:30:00.120093: I tensorflow/core/common_runtime/gpu/gpu_device.cc:847] Peer access not supported between device ordinals 0 and 1
2019-02-08 20:30:00.120205: I tensorflow/core/common_runtime/gpu/gpu_device.cc:847] Peer access not supported between device ordinals 1 and 0
2019-02-08 20:30:00.120246: I tensorflow/core/common_runtime/gpu/gpu_device.cc:976] DMA: 0 1 
2019-02-08 20:30:00.120284: I tensorflow/core/common_runtime/gpu/gpu_device.cc:986] 0:   Y N 
2019-02-08 20:30:00.120343: I tensorflow/core/common_runtime/gpu/gpu_device.cc:986] 1:   N Y 
2019-02-08 20:30:00.120384: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:06:00.0)
2019-02-08 20:30:00.120405: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:1) -> (device: 1, name: GeForce GTX TITAN X, pci bus id: 0000:81:00.0)
2019-02-08 20:30:11.669108: W tensorflow/core/common_runtime/bfc_allocator.cc:273] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.5KiB.  Current allocation summary follows.
2019-02-08 20:30:11.669210: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (256): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669233: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (512): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669251: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (1024): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669281: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (2048): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669308: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (4096): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669325: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (8192): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669342: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (16384): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669358: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (32768): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669374: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (65536): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669390: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (131072): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669407: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (262144): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669423: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (524288): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669439: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (1048576): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669455: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (2097152): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669493: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (4194304): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669513: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (8388608): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669529: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (16777216): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669545: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (33554432): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669561: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (67108864): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669577: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (134217728): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669593: I tensorflow/core/common_runtime/bfc_allocator.cc:643] Bin (268435456): 	Total Chunks: 0, Chunks in use: 0 0B allocated for chunks. 0B client-requested for chunks. 0B in use in bin. 0B client-requested in use in bin.
2019-02-08 20:30:11.669610: I tensorflow/core/common_runtime/bfc_allocator.cc:660] Bin for 1.5KiB was 1.0KiB, Chunk State: 
2019-02-08 20:30:11.669633: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114800000 of size 1280
2019-02-08 20:30:11.669648: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114800500 of size 256
2019-02-08 20:30:11.669662: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114800600 of size 256
2019-02-08 20:30:11.669677: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114800700 of size 512
2019-02-08 20:30:11.669692: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114800900 of size 1024
2019-02-08 20:30:11.669706: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114800d00 of size 256
2019-02-08 20:30:11.669720: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114800e00 of size 1536
2019-02-08 20:30:11.669735: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114801400 of size 768
2019-02-08 20:30:11.669749: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114801700 of size 256
2019-02-08 20:30:11.669774: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114801800 of size 256
2019-02-08 20:30:11.669802: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114801900 of size 9728
2019-02-08 20:30:11.669828: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114803f00 of size 512
2019-02-08 20:30:11.669854: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114804100 of size 2457600
2019-02-08 20:30:11.669879: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114a5c100 of size 1024
2019-02-08 20:30:11.669905: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Chunk at 0x8114a5c500 of size 147405568
2019-02-08 20:30:11.669930: I tensorflow/core/common_runtime/bfc_allocator.cc:693]      Summary of in-use Chunks by size: 
2019-02-08 20:30:11.669959: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 5 Chunks of size 256 totalling 1.2KiB
2019-02-08 20:30:11.669990: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 2 Chunks of size 512 totalling 1.0KiB
2019-02-08 20:30:11.670010: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 1 Chunks of size 768 totalling 768B
2019-02-08 20:30:11.670036: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 2 Chunks of size 1024 totalling 2.0KiB
2019-02-08 20:30:11.670054: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 1 Chunks of size 1280 totalling 1.2KiB
2019-02-08 20:30:11.670079: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 1 Chunks of size 1536 totalling 1.5KiB
2019-02-08 20:30:11.670096: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 1 Chunks of size 9728 totalling 9.5KiB
2019-02-08 20:30:11.670112: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 1 Chunks of size 2457600 totalling 2.34MiB
2019-02-08 20:30:11.670130: I tensorflow/core/common_runtime/bfc_allocator.cc:696] 1 Chunks of size 147405568 totalling 140.58MiB
2019-02-08 20:30:11.670147: I tensorflow/core/common_runtime/bfc_allocator.cc:700] Sum Total of in-use chunks: 142.94MiB
2019-02-08 20:30:11.670165: I tensorflow/core/common_runtime/bfc_allocator.cc:702] Stats: 
Limit:                   149880832
InUse:                   149880832
MaxInUse:                149880832
NumAllocs:                      15
MaxAllocSize:            147405568

2019-02-08 20:30:11.670183: W tensorflow/core/common_runtime/bfc_allocator.cc:277] ******************************************************xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
2019-02-08 20:30:11.670213: W tensorflow/core/framework/op_kernel.cc:1182] Resource exhausted: OOM when allocating tensor of shape [384] and type float
2019-02-08 20:30:11.670310: E tensorflow/core/common_runtime/executor.cc:644] Executor failed to create kernel. Resource exhausted: OOM when allocating tensor of shape [384] and type float
	 [[Node: local3/biases/Momentum/Initializer/zeros = Const[_class=["loc:@local3/biases"], dtype=DT_FLOAT, value=Tensor<type: float shape: [384] values: 0 0 0...>, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]
(28, 28, 96)
(28, 28, 256)
(14, 14, 256)
384
192
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1327, in _do_call
    return fn(*args)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1306, in _run_fn
    status, run_metadata)
  File "/usr/lib/python3.5/contextlib.py", line 66, in __exit__
    next(self.gen)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/errors_impl.py", line 466, in raise_exception_on_not_ok_status
    pywrap_tensorflow.TF_GetCode(status))
tensorflow.python.framework.errors_impl.ResourceExhaustedError: OOM when allocating tensor of shape [384] and type float
	 [[Node: local3/biases/Momentum/Initializer/zeros = Const[_class=["loc:@local3/biases"], dtype=DT_FLOAT, value=Tensor<type: float shape: [384] values: 0 0 0...>, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/om/user/sanjanas/eccentricity-crop/main.py", line 161, in <module>
    sess.run(tf.global_variables_initializer())
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 895, in run
    run_metadata_ptr)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1124, in _run
    feed_dict_tensor, options, run_metadata)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1321, in _do_run
    options, run_metadata)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1340, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.ResourceExhaustedError: OOM when allocating tensor of shape [384] and type float
	 [[Node: local3/biases/Momentum/Initializer/zeros = Const[_class=["loc:@local3/biases"], dtype=DT_FLOAT, value=Tensor<type: float shape: [384] values: 0 0 0...>, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]

Caused by op 'local3/biases/Momentum/Initializer/zeros', defined at:
  File "/om/user/sanjanas/eccentricity-crop/main.py", line 139, in <module>
    train_step = tf.train.MomentumOptimizer(learning_rate=lr, momentum=opt.hyper.momentum).minimize(total_loss, var_list=all_var)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/training/optimizer.py", line 325, in minimize
    name=name)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/training/optimizer.py", line 446, in apply_gradients
    self._create_slots([_get_variable_for(v) for v in var_list])
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/training/momentum.py", line 66, in _create_slots
    self._zeros_slot(v, "momentum", self._name)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/training/optimizer.py", line 766, in _zeros_slot
    named_slots[_var_key(var)] = slot_creator.create_zeros_slot(var, op_name)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/training/slot_creator.py", line 174, in create_zeros_slot
    colocate_with_primary=colocate_with_primary)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/training/slot_creator.py", line 146, in create_slot_with_initializer
    dtype)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/training/slot_creator.py", line 66, in _create_slot_var
    validate_shape=validate_shape)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/variable_scope.py", line 1065, in get_variable
    use_resource=use_resource, custom_getter=custom_getter)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/variable_scope.py", line 962, in get_variable
    use_resource=use_resource, custom_getter=custom_getter)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/variable_scope.py", line 367, in get_variable
    validate_shape=validate_shape, use_resource=use_resource)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/variable_scope.py", line 352, in _true_getter
    use_resource=use_resource)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/variable_scope.py", line 725, in _get_single_variable
    validate_shape=validate_shape)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/variables.py", line 199, in __init__
    expected_shape=expected_shape)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/variables.py", line 277, in _init_from_args
    initial_value(), name="initial_value", dtype=dtype)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/variable_scope.py", line 701, in <lambda>
    shape.as_list(), dtype=dtype, partition_info=partition_info)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/init_ops.py", line 93, in __call__
    return array_ops.zeros(shape, dtype)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/array_ops.py", line 1401, in zeros
    output = constant(zero, shape=shape, dtype=dtype, name=name)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/constant_op.py", line 106, in constant
    attrs={"value": tensor_value, "dtype": dtype_value}, name=name).outputs[0]
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 2630, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 1204, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

ResourceExhaustedError (see above for traceback): OOM when allocating tensor of shape [384] and type float
	 [[Node: local3/biases/Momentum/Initializer/zeros = Const[_class=["loc:@local3/biases"], dtype=DT_FLOAT, value=Tensor<type: float shape: [384] values: 0 0 0...>, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]

